warehouse/scripts/dbt_artifacts/manifest.py

```diff
@@ -0,0 +1,1438 @@
+    namespace: Optional[str] = None
+
+
+class ResourceType7(Enum):
+    snapshot = "snapshot"
+
+
+class SnapshotConfig(BaseModel):
+    class Config:
+        extra = Extra.allow
+
+    enabled: Optional[bool] = True
+    alias: Optional[str] = None
+    schema_: Optional[str] = Field(None, alias="schema")
+    database: Optional[str] = None
+    tags: Optional[Union[List[str], str]] = []
+    meta: Optional[Dict[str, Any]] = {}
+    group: Optional[str] = None
+    materialized: Optional[str] = "snapshot"
+    incremental_strategy: Optional[str] = None
+    persist_docs: Optional[Dict[str, Any]] = {}
+    post_hook: Optional[List[Hook]] = Field([], alias="post-hook")
+    pre_hook: Optional[List[Hook]] = Field([], alias="pre-hook")
+    quoting: Optional[Dict[str, Any]] = {}
+    column_types: Optional[Dict[str, Any]] = {}
+    full_refresh: Optional[bool] = None
+    unique_key: Optional[str] = None
+    on_schema_change: Optional[str] = "ignore"
+    grants: Optional[Dict[str, Any]] = {}
+    packages: Optional[List[str]] = []
+    docs: Optional[Docs] = Field(
+        default_factory=lambda: Docs.parse_obj({"show": True, "node_color": None})
+    )
+    contract: Optional[ContractConfig] = Field(
+        default_factory=lambda: ContractConfig.parse_obj({"enforced": False})
+    )
+    strategy: Optional[str] = None
+    target_schema: Optional[str] = None
+    target_database: Optional[str] = None
+    updated_at: Optional[str] = None
+    check_cols: Optional[Union[str, List[str]]] = None
+
+
+class ResourceType8(Enum):
+    seed = "seed"
+
+
+class SeedConfig(BaseModel):
+    class Config:
+        extra = Extra.allow
+
+    enabled: Optional[bool] = True
+    alias: Optional[str] = None
+    schema_: Optional[str] = Field(None, alias="schema")
+    database: Optional[str] = None
+    tags: Optional[Union[List[str], str]] = []
+    meta: Optional[Dict[str, Any]] = {}
+    group: Optional[str] = None
+    materialized: Optional[str] = "seed"
+    incremental_strategy: Optional[str] = None
+    persist_docs: Optional[Dict[str, Any]] = {}
+    post_hook: Optional[List[Hook]] = Field([], alias="post-hook")
+    pre_hook: Optional[List[Hook]] = Field([], alias="pre-hook")
+    quoting: Optional[Dict[str, Any]] = {}
+    column_types: Optional[Dict[str, Any]] = {}
+    full_refresh: Optional[bool] = None
+    unique_key: Optional[Union[str, List[str]]] = None
+    on_schema_change: Optional[str] = "ignore"
+    grants: Optional[Dict[str, Any]] = {}
+    packages: Optional[List[str]] = []
+    docs: Optional[Docs] = Field(
+        default_factory=lambda: Docs.parse_obj({"show": True, "node_color": None})
+    )
+    contract: Optional[ContractConfig] = Field(
+        default_factory=lambda: ContractConfig.parse_obj({"enforced": False})
+    )
+    quote_columns: Optional[bool] = None
+
+
+class MacroDependsOn(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    macros: Optional[List[str]] = []
+
+
+class ResourceType9(Enum):
```

Comment:
I find it confusing that the `ResourceType`s are just defined sporadically throughout

Reply:
yep it's a weird consequence of how they define the schemas; I actually asked someone about that, and he suggested opening up an issue to discuss